lib/components/admin-command.js

```diff
@@ -0,0 +1,157 @@
+"use strict";
+
+var minimist = require("minimist");
+
+/**
+ * Construct a new AdminCommand helper instance.
+ * @constructor
+ * @param {string} opts.desc A short description to display in the 'help'
+ * command.
+ * @param {AdminCommand~opts=} opts.opts An optional object giving commandline
+ * options recognised by the command.
+ * @param {AdminCommand~func} opts.func The handler function that implements
+ * the command.
+ */
+function AdminCommand(opts) {
+    this.desc = opts.desc;
+    this._func = opts.func;
+
+    if (opts.opts) {
+        this.optspec = {};
+        this.optaliases = {};
+
+        Object.keys(opts.opts).forEach((k) => {
+            var names = k.split(/\|/);
+
+            var result = names.shift().match(/^(!?)(.*?)$/);
+            var name = result[2];
+            var required = result[1];
+
+            this.optspec[name] = {
+                desc: opts.opts[k],
+                required: required,
+            }
+
+            names.forEach((a) => this.optaliases[a] = name);
+        });
+    }
+}
+
+/**
+ * Invoke the handler function for the command. The given list of strings will
+ * be parsed for optional arguments as per the command's opts specification, if
+ * present.
+ * @param {Object} bridge The top-level bridge instance to pass to the handler
+ * function.
+ * @param {string[]} args A list of string arguments parsed from the command
+ * line.
+ * @param {function} respond The responder function to pass to the handler
+ * function.
+ * @return {Promise} Returns the void promise returned by the handler function.
+ */
+AdminCommand.prototype.run = function(bridge, args, respond) {
+    var opts;
+    if (this.optspec) {
+        opts = minimist(args);
+
+        // Canonicalise aliases
+        Object.keys(this.optaliases).forEach((a) => {
+            if (a in opts) {
+                opts[this.optaliases[a]] = opts[a];
+                delete opts[a];
+            }
+        });
+
+        Object.keys(opts).forEach((n) => {
+            if (n === "_") return;
```

Comment:
That's a minimist thing - `_` is the special key that contains a list of the remaining positional parameters it didn't think matched any named options.